### PR TITLE
Empty Stats: Reader discover nudge

### LINKED ISSUE #17126: Empty Stats: Reader discover nudge
## Description
A nudge to comment on similar blogs to build up a network

Design ref: pbArwn-2Js-p2

## Tasks
- [x] Hook up to the Reader Discover flow: present **Follow topics** modal
- [x] Add a guided tour layer to the Follow topics modal
- [x] Once a user selects topic(s), route to the Reader's Discover tab [#17349](https://github.com/wordpress-mobile/WordPress-iOS/pull/17349)

### PR DESCRIPTION
**Pull request by nikola-milicevic**
Fixes [#17126](https://github.com/wordpress-mobile/WordPress-iOS/issues/17126)

This PR adds Reader discover nudge logic and UI.

## Test happy path
1. Go to **My Site** > **Stats** for site with less than 30 views
2. Complete or dismiss the first two nudges (**Publicize** and **Blogging reminders**) in order to have the **Reader discover** nudge shown
3. Tap on **Discover blogs to follow**
4. **Follow topics** screen should be presented
5. Select one or more topics and tap **Done**
6. **Reader** tab should be presented, with **Discover** tab selected

## Test unhappy path
1. Repeat the steps **1, 2, 3, 4** from the happy path
2. Don't select any topic and tap **Done**
3. The **Follow topics** screen should just be dismissed

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewController.swift [hunk 1]
```diff
@@ -99,6 +99,11 @@ class ReaderTabViewController: UIViewController {
 
         ReaderTracker.shared.start(.main)
     }
+
+    func presentDiscoverTab() {
+        viewModel.fetchReaderMenu()
+        viewModel.showTab(at: 1)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabViewController.swift]: What do you think of extracting 1 to a constant? Like indexForDiscover?

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift [hunk 2]
```diff
@@ -359,6 +359,13 @@ private extension SiteStatsInsightsTableViewController {
         static let cancel = NSLocalizedString("Cancel", comment: "Cancel Insight management action sheet.")
     }
 
+    // MARK: - Grow Audience Helpers
+
+    func markCurrentNudgeAsCompleted() {
+        viewModel?.markEmptyStatsNudgeAsCompleted()
+        insightsToShow = insightsToShow.filter { $0 != .growAudience }
+        refreshTableView()
+    }
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift:374]: Nice ✨

**Review by momo-ozawa — COMMENTED** [review]
Works as described, but I noticed that on first launch the Follow Topics screen includes topics that are already being followed.

1. Clean install
2. Go to Stats and dismiss other nudges to get to Reader Discover nudge
3. Tap on Discover blogs to follow
4. Make note of the tags that show up
5. Don't follow any tags
6. Go to Reader > Following > Filter > Topics and notice that some of the tags showing up in step 4 are already being followed
7. Go back to Stats
8. Tap on Discover blogs to follow
9. The correct tags are being shown now

Follow Topics, first launch | Tags that are being followed | Follow Topics, second try
-- | -- | --
Simulator Screen Shot - iPhone 12 Pro - 2021-10-22 at 10 12 52 | Simulator Screen Shot - iPhone 12 Pro - 2021-10-22 at 10 13 07 | Simulator Screen Shot - iPhone 12 Pro - 2021-10-22 at 10 13 23

**Comment by nikola-milicevic** [comment]
@momo-ozawa Thanks for checking this out. I fixed the issue with inconsistent tags on fresh install.

**Review by momo-ozawa — APPROVED** [review]
> I fixed the issue with inconsistent tags on fresh install.

Confirmed! ✅

**Pull request merged**